src/test/java/com/ita/if103java/ims/service/WarehouseServiceImplTest.java

```diff
@@ -0,0 +1,454 @@
+    private AccountType basic;
+    private UserDetailsImpl userDetails;
+    private Long accountId;
+    private Address address;
+    private AddressDto addressDto;
+
+    @Mock
+    private Warehouse warehouse;
+    @Mock
+    private WarehouseDao warehouseDao;
+    @Mock
+    private AddressDao addressDao;
+    @Mock
+    private AddressDtoMapper addressDtoMapper;
+    @Mock
+    private EventService eventService;
+    @Mock
+    private SavedItemDao savedItemDao;
+    @Mock
+    private WarehouseDtoMapper warehouseDtoMapper;
+    @Mock
+    private SavedItemService savedItemService;
+
+    @InjectMocks
+    private WarehouseServiceImpl warehouseService;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.initMocks(this);
+        basic = new AccountType();
+        basic.setMaxWarehouses(3);
+        basic.setMaxWarehouseDepth(3);
+        basic.setId(1L);
+        User user = new User();
+        user.setId(1L);
+        user.setAccountId(1L);
+        user.setRole(Role.ROLE_WORKER);
+        userDetails = new UserDetailsImpl(user);
+        userDetails.setAccountType(basic);
+        topWarehouseDto = new WarehouseDto(1L, "WarehouseTop", "auto parts", 0, false, null, 1L, null, true, null);
+        warehouseDto = new WarehouseDto(12L, "WarehouseTest", "auto parts", 20, true, 5L, 1L, 4L, true, null);
+        address = new Address("Ukraine", "Kyiv", "Stusa, 5", "77000", 48F, 50F);
+        addressDto = new AddressDto(1L, "Ukraine", "Kyiv", "Stusa, 5", "77000", 48F, 50F);
+    }
+
+    @Test
+    public void add_successFlow() {
+        int quantity = 2;
+        int level = 0;
+        int maxQuantity = userDetails.getAccountType().getMaxWarehouses();
+        Warehouse warehouseCreate = new Warehouse(1L, "WarehouseTop", "auto parts", 0, false, null, 1L, null, true);
+        warehouseDto.setParentID(null);
+        Long accountId = topWarehouseDto.getAccountID();
+        assertNull(topWarehouseDto.getParentID());
+        when(warehouseDao.findQuantityOfWarehousesByAccountId(accountId)).thenReturn(quantity);
+        assertTrue(quantity < maxQuantity);
+        when(warehouseDao.create(warehouseDtoMapper.toEntity(warehouseDto))).thenReturn(warehouseCreate);
+        when(addressDtoMapper.toEntity(warehouseDto.getAddressDto())).thenReturn(address);
+        assertTrue(warehouseCreate.isTopLevel());
+        when(addressDao.createWarehouseAddress(warehouseCreate.getId(), address)).thenReturn(address);
+        when(addressDtoMapper.toDto(address)).thenReturn(addressDto);
+
+        Event event = new Event("Warehouse created " +
+            warehouse.getName(), accountId,
+            warehouse.getId(), 1L, EventName.WAREHOUSE_CREATED, 2L);
+        doNothing().when(eventService).create(event);
+//      verify(eventService, times(1)).create(event);
+//         //form a return DTO
+        when(warehouseDtoMapper.toDto(warehouseCreate)).thenReturn(topWarehouseDto);
+        topWarehouseDto.setAddressDto(addressDto);
+
+    }
```

Comment:
this test test nothing, there is no invocation of the warehouseService.add
Fix or remove